apps/main/lib/main/repositories/articles_repo.rb

```diff
@@ -1,8 +1,11 @@
 require "blog/repository"
+require "main/entities/article"
 
 module Main
   module Repositories
     class ArticlesRepo < Blog::Repository[:articles]
+      struct_namespace Main::Entities
+
```

Comment:
@choonkeat Doing the `require` like this _technically_ required, but I did it for 2 purposes:

1. We need to ensure the `Main::Entities` constant exists before we can refer to it as the struct namespace. Requiring `"main/entities/article"` ensures that this module has been defined.
2. Using `"main/entities/article"` as the require also kind of acts as a "hint" that this is the entity object we're likely to be returning from this repo.

So that's why I added it.

An alternative approach would've been something like this, defining the module inline.

```ruby
module Main
  module Entities
  end

  module Repositories
    class ArticlesRepo < Blog::Repository[:articles]
      struct_namespace Main::Entities
```

Or perhaps we could also create a single file with an empty `Entities` module so we could just require that instead of a particular entity, e.g.

```ruby
require "main/entities"

module Main
  module Repositories
    class ArticlesRepo < Blog::Repository[:articles]
      struct_namespace Main::Entities
```

Now that I think about it, this feels a little nicer.

What I _also_ expect to happen is that, once we want to build a second repository in the Main sub-app, I'll extract the `struct_namespace` declaration out into a shared `Main::Repository` superclass, which can be where we think about adjusting how we ensure `Main::Entities` exists.

One last thing – all of this has been because in a dry-web app, one principle we follow is that _any_ file should be able to be loaded on its own and run, without any of the rest of the system having been loaded first. So that means each file needs to be responsible for its requires and everything else.